### PR TITLE
Bugfix: Inverted mask in `xformers` rotary attention forward

### PR DESCRIPTION
**Pull request by vinamarora8**
The mask to bias transformation in `rotary_attn_xformers_func` is currently the opposite of what it should be.
The bug only surfaces if you have `xformers` installed in your environment. If you're using pytorch attention, you would be unaffected.

### CODE REVIEW

**Review by mazabou — APPROVED** [review]
Thanks @vinam for fixing this. Did you try to run with xformers to confirm that the outcome would be the exact same as the one with pytorch? I am also happy to test this if not.

**Comment by vinamarora8** [comment]
> Thanks @vinam for fixing this. Did you try to run with xformers to confirm that the outcome would be the exact same as the one with pytorch? I am also happy to test this if not.

Just tested ~50 epochs on 2 sessions of MP. It's not _exactly_ the same; I would attribute it to different numerical precision in the two kernels. However, it atleast trains with this fix. It wasn't learning at all when the mask was flipped (train loss wouldn't go down, as you would expect)

image
